modeshape-jcr/src/main/java/org/modeshape/jcr/value/binary/MongodbBinaryStore.java

```diff
@@ -192,8 +199,10 @@ public BinaryValue storeValue( InputStream stream,
 
             // store content
             DBCollection content = db.getCollection(key.toString());
-            ChunkOutputStream dbStream = markAsUnused ? new ChunkOutputStream(content, System.currentTimeMillis()) : new ChunkOutputStream(
-                                                                                                                                           content);
+            content.createIndex(DATA_CHUNK_SORT_INDEX);
```

Comment:
each binary value collection must have its own index, so that's the reason this is called each time
note that `db.getCollection(key.toString());` will create a new collection for each binary value

I don't know the internals of how Mongo manages Indexes on collections, but I would not try doing this in the background because that would mean adding synchronization between creating a new binary value and reading that binary value, in the sense that until the index is created one cannot read the value back. This complexity is simply not worth it IMO.